main/boost/APKBUILD

```diff
@@ -154,28 +165,42 @@ package() {
 }
 
 static() {
-	pkgdesc="Static libraries for boost"
+	pkgdesc="Boost static libraries"
+	depends="$depends_static"
+
 	mkdir -p "$subpkgdir"/usr/lib
-	mv "$pkgdir"/usr/lib/*.a "$subpkgdir"/usr/lib/
+	mv "$pkgdir"/usr/lib/lib*.a "$subpkgdir"/usr/lib/
 }
 
 _boostlib() {
 	local name="${subpkgname#$pkgname-}"
-	pkgdesc="Boost $name library"
-	case "$name" in
-	python*) depends="$depends $name"
-	esac
+	pkgdesc="Boost $name shared library"
+	depends="$depends__boostlib"
 
 	mkdir -p "$subpkgdir"/usr/lib
-	mv "$pkgdir"/usr/lib/libboost_$name* "$subpkgdir"/usr/lib/
+	mv "$pkgdir"/usr/lib/libboost_$name*.so.[0-9]* "$subpkgdir"/usr/lib/
+}
+
+libs() {
+	default_libs
+
+	pkgdesc="Boost shared libraries"
+	depends="$depends_libs"
+	local _pkg; for _pkg in $origsubpackages; do
+		case "$_pkg" in
+			*:_boostlib) depends="$depends ${_pkg%%:*}" ;;
+		esac
+	done
+
+	mkdir -p "$subpkgdir"
```

Comment:
I still think the `boost-libs` can be removed. Do you really need it or can you use the `apk add --virtual ...` trick?

I want keep things simple if possible.

Reply:
I don't understand the reason for removing `boost-libs` when it gives good default behavior for the users at very little cost. There may be an argument for removing the dependency on `boost-libs` from `boost`. However, I do rather like the idea that the user doesn't need to know anything about how we chose to split this package to install a working set of runtime files.

The `scanelf` trick is useful, but encoding the libs list in `boost-libs` is the right choice in my opinion.